### PR TITLE
Documentation/tictactoe

### PR DESCRIPTION
**Pull request by owulveryck**
If you have a couple of minutes @chewxy, I'd like a quick review of the README to check if all the writing makes sense.

It is not crystal clear why both players need to `SwitchToInference` yet, but it seems to work.

Besides that, this PR :

- adds comments for the godoc 
- moves the `gtp` and `online` packages into `internal`

### CODE REVIEW

File: agent.go [hunk 1]
```diff
@@ -35,7 +36,7 @@ func newAgent(a Dualer) *Agent {
 	return retVal
 }
 
-// SwitchTOInference uses the inference mode neural network.
+// SwitchToInference uses the inference mode neural network.
```
Comment (owulveryck) [inline agent.go:39]: What does this function do under the hood?

File: README.md [hunk 2]
```diff
@@ -1,4 +1,187 @@
...
+- Creating a _configuration_ for your AZ internal MCTS and NN.
+- Creating an `AZ` structure based on the _game_ and  the _configuration_
+- Executing the learning process (by calling the [`Learn`](https://pkg.go.dev/github.com/gorgonia/agogo#AZ.Learn) method)
+- Saving the trained model (by calling the [`Save`](https://pkg.go.dev/github.com/gorgonia/agogo#AZ.Save) method)
+
+The steps to play against the algorithm are:
+
+- Creating an `AZ` object
+- Loading the trained model (by calling the [`Read`](https://pkg.go.dev/github.com/gorgonia/agogo#AZ.Read) method)
+- Get the AI move by calling the [`Search`](https://pkg.go.dev/github.com/gorgonia/agogo#Agent.Search) method and applying the move to the game manually
```
Comment (owulveryck) [inline README.md:44]: Maybe we should document the `SwithToInference` here?

File: agent.go (hunk 1 above)
Comment (chewxy) [reply agent.go:39]: `BatchNorm` "knows" about whether it's in training mode or inference mode. That's why this is needed.

**Comment by chewxy** [comment]
If this is good to merge you can merge it. It looks good to me. thank you so much for documenting this.

**Pull request merged**